app/controllers/ads_controller.rb

```diff
@@ -0,0 +1,18 @@
+class AdsController < ApplicationController
+  def index; end
+
+  def new
+    api = Api.new("categories")
+    @api = api.response["categories"]
+    @ad = Ad.new
+  end
+
+  def create
+    @ad = Ad.create(ad_params)
```

Comment:
do you really need an instance variable here? i think it can be  just a local varibale.
also, it might be that the Ad is not saved, and it might throw an error. try to handle it, for example,
```ruby
ad = Ad.new(ad_params)
if ad.save
  do this
else
  do that
end
```